reclaim.c

```diff
@@ -457,8 +457,13 @@ STATIC void GC_reclaim_block(struct hblk *hbp, word report_if_found)
           GC_ASSERT(sz * hhdr -> hb_n_marks <= HBLKSIZE);
 #       endif
         if (report_if_found) {
-          GC_reclaim_small_nonempty_block(hbp, sz,
-                                          TRUE /* report_if_found */);
+#         if defined(__CHERI_PURE_CAPABILITY__)
+            GC_reclaim_small_nonempty_block(hhdr->hb_block, sz,
+                                            TRUE /* report_if_found */);
```

Comment:
would it be neater to move the TRUE outside the #ifdef CHERI ? avoiding code duplication?

Reply:
I personally think that this makes the code look a bit disjoint as a single TRUE appears in a line all by itself.  But if that is consensus, I don't mind changing it.

Reply:
Another possibility might be something like:

```c
#if CHERI
blk = hhdr->hb_block
#else
blk = hbp
#endif
GC_reclaim_small_nonempty_block(blk, sz, TRUE);
```

?

Reply:
Looks neater @ltratt . Will change in the next commit.

Reply:
> Please squash. [I guess you're going to break the tabs/spaces stuff out into a second commit?]

The mistakes, I will push through in a different commit.